tasks/common.py

```diff
@@ -13,22 +14,28 @@
 from log_utils import get_logger
 
 REPO_BASE_URL = 'github.com'
-WORKING_DIR = path.join(path.sep, 'tmp')
+
+# We use the `/tmp` folder to hold the cloned, and later the built,
+# client site and repository. Using a predictable path in our
+# ephemeral build containers is ok and does not present any
+# security issues because no sensitive information is stored there;
+# it only holds the client site code. Thus, marked as nosec.
```

Comment:
@commit-dkp: want to check my math here? `bandit` was picking up use of `/tmp`, but I don't see an issue with it for our use case.

Reply:
@jseppi Would be less risky to use https://docs.python.org/3/library/tempfile.html#tempfile.mkdtemp .  Any reason not to?

Reply:
I think I had trouble when I initially tried that because of how PyInvoke works (or due to how I'm using it). Basically I was losing the reference to the created temp directory between tasks.

What's the risk?

Reply:
@jseppi Interesting, and happy to dive into it.  The risk to Federalist is that since `/tmp` is world-writable, someone else might create in it the directories Federalist will use, and put in them their own malicious files to add to the builds.  Using mkdtemp reduces that risk by safely creating a unique tmp directory on demand that is read-write restricted to Federalist.  Have you tried using https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory instead of mkdtemp?

Reply:
I don't think `TemporaryDirectory` would help either. Because the sub-tasks essentially get run in separate processes, the `common` module gets reloaded for each one, and thus a randomly created tmpdir would be created anew, but each task needs to know where the files from the previous task live.

What if I just use something other than `/tmp`, like a new directory called `/work_dir`?

Reply:
@jseppi Gotcha.  Sounds, then, to me like you need a not-so-temporary directory for the persistence between tasks.  Assuming that `/` is only writable by `root`, then a `/work_dir` that is only read-writeable by the user Federalist runs as would be equally effective; an other, non-root user would not be able to squat on that directory.

Reply:
Everything is run as `root` in this container. Should there be a separate that runs the build process? The build containers are ephemeral and exist only for a single site build at a time.

Reply:
@jseppi In general, processes shouldn't have more privileges than they need, but we can take that up separately.  It's enough to resolve this question that `/work_dir` could only be squatted on by another superuser, which is effectively the same level of protection afforded by TemporaryDirectory/mkdtemp.